src/network/connectionpool.py

```diff
@@ -26,9 +27,24 @@
 
 
 @Singleton
-# pylint: disable=too-many-instance-attributes
 class BMConnectionPool(object):
     """Pool of all existing connections"""
+    # pylint: disable=too-many-instance-attributes
+
+    trustedPeer = None
+    """
+    If the trustedpeer option is specified in keys.dat then this will
+    contain a Peer which will be connected to instead of using the
+    addresses advertised by other peers. The client will only connect to
+    this peer and the timing attack mitigation will be disabled in order
+    to download data faster. The expected use case is where the user has
+    a fast connection to a trusted server where they run a Bitmessage
+    daemon permanently. If they then run a second instance of the client
+    on a local machine periodically when they want to check for messages
+    it will sync with the network a lot faster without compromising
+    security.
```

Comment:
I find it unlikely that the performance difference will be important. First, download happens asynchronously and only stalls if the decode buffer (32MB) gets full. Second, if an object isn't for you, this performance optimisation makes no difference. So if almost none of the objects are for you (as it is normally), there is no difference. Third, if you have a lot of addresses, assuming a random order, it still needs to process at least 50% on average.

So, in total, you will save 50% of time for processing objects that are for you, and only download slower to the extent that this affects a large processing buffer.

I think the risk of introducing potential attack vectors isn't worth it.

Reply:
Well, as you probably see, it is not originally my docstring. I moved it from `state`, but its first appearance is in `shared` in 0326315.

Reply:
That refers to an older implementation and a different aspect. It used delays while downloading, while now it downloads as quickly as the CPU allows. This was prior to moving to asyncore. Even further in the past the downloading and decryption were in the same thread so the delays made sense, it probably was an artifact of this, but that was even before I started contributing.

The change you made changes the "constant time decryption" approach I introduced in 9239813ebb. It probably still isn't constant as it still skips AES decryption on a failure, but I think it's still an improvement as AES is very fast compared to ECC.